mitodl/mitxpro
# Added basic course catalog

#### Pre-Flight checklist


#### What are the relevant tickets?
Closes #34 

#### What's this PR do?
- Adds the course catalog (implemented as a Django template)
- Sets the root path of the app to render the course catalog

#### How should this be manually tested?
Run the seed data command (`manage.py seed_data`) and visit the root path. You should see cards for programs and courses.

#### Any background context you want to provide?
- This page renders a card for programs _and_ for all courses (whether they are part of a program or not). For this first implementation there isn't really any special ordering going on besides rendering the program cards before the course cards
- Based on https://mitxpro.mit.edu/courses (in absence of formal designs)
- For this first implementation I decided to leave off the links that you see in the live xpro site linked above. Those links are for individual course runs (our seed data course runs don't have anything to link to yet) and we don't yet have course/program detail pages that we could link to otherwise. These links are probably best to implement in another PR
- I included a default image for courses/programs that don't have an image set, but in the future we will probably make this a required field
- Even though this page is rendered with a Django template and has no interactive elements, I included the full JS bundle in the page for things like GA and Raven

#### Screenshots (if appropriate)
![ss 2019-03-15 at 15 39 20 ](https://user-images.githubusercontent.com/14932219/54457872-f1ae8d80-4738-11e9-97db-28241eb3ecfc.png)
![ss 2019-03-15 at 15 39 33 ](https://user-images.githubusercontent.com/14932219/54457877-f3785100-4738-11e9-9174-a569a1d6efed.png)
![ss 2019-03-15 at 15 39 42 ](https://user-images.githubusercontent.com/14932219/54457881-f4a97e00-4738-11e9-9ac9-e3bd887a00b0.png)
